android/src/main/java/org/celo/BlindThresholdBlsModule.java

```diff
@@ -36,24 +37,38 @@ public String getName() {
         return "BlindThresholdBls";
     }
 
-    // TODO add support for multilpe outstanding blind calls
+    // Blind the message
+    // @randomness allows caller to provide deterministic form of randomness
     @ReactMethod
-    public void blindMessage(String message, Promise promise) {
+    public void blindMessageWithRandom(String message, String randomness, Promise promise) {
         try {
             Log.d(TAG, "Preparing blind message buffers");
             byte[] messageBytes = Base64.decode(message, Base64.DEFAULT);
             messageBuf = new Buffer(messageBytes);
             Buffer blindedMessageBuf = new Buffer();
 
-            Log.d(TAG, "Preparing blinding seed");
-            Random random = new Random();
-            byte[] seed = new byte[32];
-            random.nextBytes(seed);
-            Buffer seedBuf = new Buffer(seed);
+            byte[] randomnessBytes = Base64.decode(randomness, Base64.DEFAULT);
+
+            // Must be 32 length
+            int originalLength = randomnessBytes.length;
+            randomnessBytes = Arrays.copyOf(randomnessBytes, 32);
+
+            // Pad if not enough bytes
+            if (originalLength < 32) {
```

Comment:
Can you confirm we want to pad with random bits in this case instead of just error?

Reply:
Yeah, thanks for raising this. I'll remove the padding:
1. To be consistent with objective-c implementation
2. Prevent it from being non-deterministic when caller expects it to be deterministic.